<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <arg name="yolo_model" default="yolov8n-pose.pt"/>
  <arg name="detection_topic" default="detection_result"/>
  <arg name="image_topic" default="/camera/rgb/image_color/compressed"/> <!-- Images in the compressed format --> 
  <arg name="conf_thres" default="0.25"/>
  <arg name="iou_thres" default="0.45"/>
  <arg name="max_det" default="300"/>
  <arg name="tracker" default="bytetrack.yaml"/>
  <arg name="debug" default="true"/>
  <arg name="debug_conf" default="true"/>
  <arg name="debug_line_width" default="1"/>
  <arg name="debug_font_size" default="1"/>
  <arg name="debug_font" default="Arial.ttf"/>
  <arg name="debug_labels" default="true"/>
  <arg name="debug_boxes" default="true"/>
  <arg name="bag_file_path" default="$(find ultralytics_ros)/KTP_dataset_bags/bags/Translation.bag"/> <!--The location of a ROS bag -->  
  
  <!-- Object detection node -->
  <node name="tracker_node" pkg="ultralytics_ros" type="tracker_node_com.py" output="screen">
  <!--<node name="tracker_node" pkg="ultralytics_ros" type="tracker_node_bag.py" output="screen">-->
    <param name="yolo_model" value="$(arg yolo_model)"/>
    <param name="detection_topic" value="$(arg detection_topic)"/>
    <param name="image_topic" value="$(arg image_topic)"/>
    <param name="conf_thres" value="$(arg conf_thres)"/>
    <param name="iou_thres" value="$(arg iou_thres)"/>
    <param name="max_det" value="$(arg max_det)"/>
    <param name="tracker" value="$(arg tracker)"/>
    <param name="debug" value="$(arg debug)"/>
    <param name="debug_conf" value="$(arg debug_conf)"/>
    <param name="debug_line_width" value="$(arg debug_line_width)"/>
    <param name="debug_font_size" value="$(arg debug_font_size)"/>
    <param name="debug_font" value="$(arg debug_font)"/>
    <param name="debug_labels" value="$(arg debug_labels)"/>
    <param name="debug_boxes" value="$(arg debug_boxes)"/>  
  </node>
  <!-- Image view -->
  <node if="$(arg debug)" name="image_view" pkg="image_view" type="image_view" output="screen">
    <remap from="image" to="debug_image"/>
  </node>
  
  <!-- ROS bag player -->  
  <node name="bag_player" pkg="rosbag" type="play" output="screen" args="$(arg bag_file_path)"/> <!-- Play back the content of a ROS bag --> 
</launch>

